File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    unordered_map<string, int> mp;
    for(string s : participant){
        mp[s]++;
    }
    
    for(string s : completion){
        mp[s]--;
    }
    
    for(auto a : mp){
        if(a.second == 1){
            answer = a.first;
            break;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

public class Solution {
    public String solution(String[] participant, String[] completion) {
        
        String answer = "";
        Map<String, Integer> myMap = new HashMap<>();

        // 참가자 목록을 맵에 추가
        for (String s : participant) {
            myMap.put(s, myMap.getOrDefault(s, 0) + 1);
        }

        // 완주자 목록을 맵에서 제거
        for (String s : completion) {
            myMap.put(s, myMap.get(s) - 1);
        }

        // 완주하지 못한 참가자 찾기
        for (String s : participant) {
            if (myMap.get(s) > 0) {
                answer = s;
                break;
            }
        }

        return answer;
    }
}